src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c

```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11538.s")
+    func_80C12B1C(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actor.world.rot.x = -this->actor.shape.rot.x;
+    if (this->actor.colChkInfo.health != 0) {
+        Actor_SetVelocityAndMoveXYRotation(&this->actor);
+    } else {
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11590.s")
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 25.0f, 25.0f, 50.0f, 7);
+    if (this->actionFunc == func_80C1193C) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C118E4.s")
+    if (this->collider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C1193C.s")
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    if (this->unk_3D4 > 0.0f) {
+        if (this->unk_18C != 10) {
+            Math_StepToF(&this->unk_3D4, 0.0f, 0.05f);
+            this->unk_3D8 = (this->unk_3D4 + 1.0f) * 0.25f;
+            this->unk_3D8 = CLAMP_MAX(this->unk_3D8, 0.5f);
+        } else if (!Math_StepToF(&this->unk_3DC, 0.5f, 0.0125f)) {
+            func_800B9010(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11C60.s")
+    func_80C12D00(this);
+    if (((this->skelAnime.animCurrentSeg == &D_06000604) && func_801378B8(&this->skelAnime, 13.0f)) ||
+        ((this->skelAnime.animCurrentSeg == &D_06000278) && func_801378B8(&this->skelAnime, 1.0f))) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_KAICHO_FLUTTER);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11D14.s")
+s32 EnThiefbird_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                 Actor* thisx) {
+    EnThiefbird* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11DC0.s")
+    if ((limbIndex == 10) || (limbIndex == 11)) {
+        this->unk_3E4 = *dList;
+        *dList = NULL;
+    } else if (limbIndex == 16) {
+        *dList = NULL;
+    } else if (limbIndex == 8) {
+        rot->z += this->unk_194;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11DF0.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11F6C.s")
+void EnThiefbird_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static s8 D_80C13698[] = {
+        -1, 0, -1, 1, 3, -1, 2, 5, -1, -1, 7, 8, 9, -1, -1, 10, -1, 0, 0, 0,
+    };
```

Comment:
```suggestion
    static s8 D_80C13698[] = {
        -1, 0, -1, 1, 3, -1, 2, 5, -1, -1, 7, 8, 9, -1, -1, 10, -1,
    };
```
17 limbs, so the last 3 `0`s are pad